docs/changelog/114207.yaml

```
@@ -0,0 +1,15 @@
+pr: 114207
+summary: Remove `cluster.routing.allocation.disk.watermark.enable_for_single_data_node` setting
+area: Allocation
+type: breaking
+issues: []
+breaking:
+  title: Remove `cluster.routing.allocation.disk.watermark.enable_for_single_data_node` setting
+  area: 'Cluster and node setting'
+  details: >-
+    Prior to 7.8, whenever a cluster had only a single data node, the watermarks would not be respected.
+    This was incompatible with how storage based autoscaling works and in order to change this in 7.8+ in a backwards
+    compatible way, we had to introduce the `cluster.routing.allocation.disk.watermark.enable_for_single_data_node`
+    node setting. The setting was deprecated in 7.14 and was made to accept only true in 8.0
```

I think there is no need to repeat why we did this, that is long gone history.

```suggestion
  details: >-
    Prior to 7.8, whenever a cluster had only a single data node, the watermarks would not be respected.
    In order to change this in 7.8+ in a backwards compatible way, we introduced the 
    `cluster.routing.allocation.disk.watermark.enable_for_single_data_node`
    node setting. The setting was deprecated in 7.14 and was made to accept only true in 8.0
```